### PR TITLE
dns: add support for /v2/quotas

### LINKED ISSUE #2398: Add support for ``dns/v2/quotas`` and ``dns/v2/recordsets``
Before starting a PR, please read our [contributor tutorial](https://github.com/gophercloud/gophercloud/tree/master/docs/contributor-tutorial).

### PR DESCRIPTION
**Pull request by SuperSandro2000**
Prior to starting a PR, please make sure you have read our
[contributor tutorial](https://github.com/gophercloud/gophercloud/tree/master/docs/contributor-tutorial).

Prior to a PR being reviewed, there needs to be a Github issue that the PR
addresses. Replace the brackets and text below with that issue number.

Fixes #2398

Links to the line numbers/files in the OpenStack source code that support the
code in this PR:

https://docs.openstack.org/api-ref/dns/#view-quotas

### CODE REVIEW

**Comment by mandre** [comment]
Hey @SuperSandro2000 , thanks for the contribution. Could you also add acceptance tests please? They are important to us because they allow to verify the code works correctly on all supported OpenStack versions and catch regressions.

**Comment by EmilienM** [comment]
For reference I'm trying to bump Coveralls to see if that makes a difference, in all our PRs the job is never happy and reports red if coverage didn't change.
Indeed, 0.04% isn't a big deal.

**Comment by SuperSandro2000** [comment]
You should be able to configure that on the mostly undocumented settings page https://coveralls.io/github/gophercloud/gophercloud/settings

**Comment by EmilienM** [comment]
I'm not admin on this page. @jtopjian could help probably?

**Comment by jtopjian** [comment]
@EmilienM Just to confirm: did you sign in to the Coveralls page via Github authentication? Once I did that, I was able to access the settings.

If that's still not working, let me know what percentage you'd like me to change the total coverage threshold and coverage decreased threshold.

**Comment by EmilienM** [comment]
> @EmilienM Just to confirm: did you sign in to the Coveralls page via Github authentication? Once I did that, I was able to access the settings.
> 
> If that's still not working, let me know what percentage you'd like me to change the total coverage threshold and coverage decreased threshold.

I have access! Seems like I didn't look in the right place :-)
https://coveralls.io/github/gophercloud/gophercloud/settings

File: acceptance/openstack/dns/v2/quotas.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+	"github.com/gophercloud/gophercloud/acceptance/tools"
+	"github.com/gophercloud/gophercloud/openstack/dns/v2/quotas"
+	th "github.com/gophercloud/gophercloud/testhelper"
+)
+
+func TestSchedulerStatsList(t *testing.T) {
+	client, err := clients.NewDNSV2Client()
+	th.AssertNoErr(t, err)
+
+	randomUUID := "513788b0-4f1b-4107-aee2-fbcdca9b9833"
```
Comment (mandre) [inline acceptance/openstack/dns/v2/quotas.go:19]: This is likely going to generate an error, querying for a project that doesn't exist.

I'd expect we validate both Get and Update() functions.

You could for instance request the current quota, modify a value, then check it was effectively modified.
Then, if you implement quota reset, you can restore the original quota and verify it was changed back to the initial value.

You can use `clients.RequireAdmin(t)` to request admin creds if needed.

**Review by mandre — CHANGES_REQUESTED** [review]
We need a bit more work on the acceptance tests before we can merge. I left some comments, let me know if something is unclear.

Also, would be cool to implement quota reset too: https://docs.openstack.org/api-ref/dns/?expanded=#reset-quotas

**Comment by SuperSandro2000** [comment]
I currently don't have time to add the reset-quotas endpoint and do a full end to end test.

**Comment by pierreprinetti** [comment]
Being reworked in #3186

**Pull request closed without merging**